worker/pkg/workflows/datasync/activities/activities_test.go

```diff
@@ -310,6 +310,154 @@ func Test_buildProcessorMutation(t *testing.T) {
 	})
 	assert.Error(t, err)
 	assert.Empty(t, output)
+
+}
+
+func Test_buildProcessorConfigJavascript(t *testing.T) {
+	mockTransformerClient := mgmtv1alpha1connect.NewMockTransformersServiceClient(t)
+	mockJobClient := mgmtv1alpha1connect.NewMockJobServiceClient(t)
+	mockConnectionClient := mgmtv1alpha1connect.NewMockConnectionServiceClient(t)
+
+	pgcache := map[string]pg_queries.DBTX{
+		"fake-prod-url":  pg_queries.NewMockDBTX(t),
+		"fake-stage-url": pg_queries.NewMockDBTX(t),
+	}
+	pgquerier := pg_queries.NewMockQuerier(t)
+	mysqlcache := map[string]mysql_queries.DBTX{}
+	mysqlquerier := mysql_queries.NewMockQuerier(t)
+
+	bbuilder := newBenthosBuilder(pgcache, pgquerier, mysqlcache, mysqlquerier, mockJobClient, mockConnectionClient, mockTransformerClient)
+	ctx := context.Background()
+
+	mockTransformerClient.On(
```

Comment:
This test could be a lot simpler if it instead just defined a standard javascript system transformer instead of a user defined one. One less thing to mock.
We have other tests that should be testing the user defined transformer.